mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityHelperGetMethodGenerator.java

```diff
@@ -56,84 +58,123 @@ public EntityHelperGetMethodGenerator(
         String childHelper = enclosingClass.addEntityHelperField(entityClass);
-        getBuilder.addStatement(
-            "$T $L = $L.get($L)", entityClass, valueName, childHelper, udtValueName);
-        getBuilder.addStatement("returnValue.$L($L)", setterName, valueName);
+        getBuilder.addStatement("$L = $L.get($L)", propertyValueName, childHelper, udtValueName);
         getBuilder.endControlFlow();
       } else {
         // Collection of other entity class(es): the CQL column is a collection of mapped UDTs
         // Build a copy of the value, decoding all UdtValue instances into entities on the fly.
-        String mappedCollectionName = enclosingClass.getNameIndex().uniqueField("mappedCollection");
+        //     CollectionTypeT propertyValue1;
+        //     RawCollectionTypeT rawCollection1 = source.get("column", GENERIC_TYPE);
+        //     if (rawCollection1 == null) {
+        //       propertyValue1 = null;
+        //     } else {
+        //       traverse rawCollection1 and convert all UdtValue into entity classes, recursing
+        //       into nested collections if necessary
+        //     }
+        getBuilder.addStatement("$T $L", type.asTypeName(), propertyValueName);
+
         String rawCollectionName = enclosingClass.getNameIndex().uniqueField("rawCollection");
         TypeName rawCollectionType = type.asRawTypeName();
+        getBuilder.addStatement(
+            "$T $L = source.get($L, $L)",
+            rawCollectionType,
+            rawCollectionName,
+            cqlName,
+            enclosingClass.addGenericTypeConstant(rawCollectionType));
+
         getBuilder
-            .addStatement(
-                "$T $L = source.get($L, $L)",
-                rawCollectionType,
-                rawCollectionName,
-                cqlName,
-                enclosingClass.addGenericTypeConstant(rawCollectionType))
-            .beginControlFlow("if ($L != null)", rawCollectionName);
-        convertUdtsIntoEntities(rawCollectionName, mappedCollectionName, type, getBuilder);
-        getBuilder
-            .addStatement("returnValue.$L($L)", setterName, mappedCollectionName)
-            .endControlFlow();
+            .beginControlFlow("if ($L == null)", rawCollectionName)
+            .addStatement("$L = null", propertyValueName)
+            .nextControlFlow("else");
+        convertUdtsIntoEntities(rawCollectionName, propertyValueName, type, getBuilder);
+        getBuilder.endControlFlow();
+      }
+
+      if (mutable) {
+        getBuilder.addStatement("$L.$L($L)", resultName, setterName, propertyValueName);
+      }
+    }
+
+    if (mutable) {
+      // We've already created an instance and filled the properties as we went
+      getBuilder.addStatement("return returnValue");
+    } else {
+      // Assume an all-arg constructor exists, and call it with all the temporary variables
+      getBuilder.addCode("$[return new $T(", returnType);
```

Comment:
1. Yes. I was going to say that the validation error will manifest as a compile error as well, but it will be clearer since we can issue it on the entity type with a more precise message. I'll add that to the entity factory (and also check for a no-arg constructor for mutable entities while we're at it).
2. Not a definitive decision, but I was going for order of declaration, as parsed by the entity factory. Matching by type sounds a bit complicated.

Reply:
Is the parsing order fully deterministic? And what is that order in the presence of type inheritance?